### PR TITLE
operator/ingress: Add NodePort publishing strategy

### PR DESCRIPTION
**Pull request by ironcladlou**
This commit implements the NodePort publishing strategy API described in the
`ingress-nodeport-publishing` enhancement proposal[1]. Please refer to the
enhancement proposal for more details.

[1] https://github.com/openshift/enhancements/blob/master/enhancements/network/ingress-nodeport-publishing.md

### CODE REVIEW

File: operator/v1/types_ingress.go [hunk 1]
```diff
@@ -188,6 +188,9 @@ const (
 
 	// Private does not publish the ingress controller.
 	PrivateStrategyType EndpointPublishingStrategyType = "Private"
+
+	// NodePort publishes the ingress controller using a Kubernetes NodePort Service.
+	NodePortStrategyType EndpointPublishingStrategyType = "NodePort"
```
Comment (Miciah) [inline operator/v1/types_ingress.go]: `NodePortStrategyType`→`NodePortServiceStrategyType` for consistency with `LoadBalancerServiceStrategyType`?

Comment (ironcladlou) [reply operator/v1/types_ingress.go]: Should the name be NodePortService everywhere instead of NodePort?

Comment (Miciah) [reply operator/v1/types_ingress.go]: To be consistent, yeah.

Comment (ironcladlou) [reply operator/v1/types_ingress.go]: Updated throughout for consistency with LoadBalancerService.

Comment (Miciah) [reply operator/v1/types_ingress.go]: Well, I take that partly back: We use the name `LoadBalancerStrategy` for the union type for the `LoadBalancerService` strategy type, so perhaps we should use the name `NodePortStrategy` for the new union type.

Comment (ironcladlou) [reply operator/v1/types_ingress.go]: Updated to be consistent with LoadBalancer's inconsistencies 😁

**Comment by ironcladlou** [comment]
@smarterclayton @openshift/sig-network-edge PTAL

**Pull request merged**